pkg/sources/udsource/metrics.go

```diff
@@ -23,13 +23,6 @@ import (
 	"github.com/numaproj/numaflow/pkg/metrics"
 )
 
-// udsourceReadCount is used to indicate the number of messages read
-var udsourceReadCount = promauto.NewCounterVec(prometheus.CounterOpts{
-	Subsystem: "udsource",
-	Name:      "read_total",
-	Help:      "Total number of messages Read",
-}, []string{metrics.LabelVertex, metrics.LabelPipeline, metrics.LabelVertexReplicaIndex, metrics.LabelPartitionName})
-
 // udsourcePending is used to indicate the number of messages pending in the user defined source
 var udsourcePending = promauto.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
same as above, can pending also be generalized?

Reply:
https://github.com/numaproj/numaflow/pull/2207#discussion_r1851336667